static/app/components/performance/teamKeyTransaction.tsx

```diff
@@ -135,90 +137,168 @@ type SelectorProps = {
   counts: Map<string, number>;
 };
 
-function TeamKeyTransactionSelector({
-  title: Title,
-  handleToggleKeyTransaction,
-  teams,
-  counts,
-  keyedTeams,
-}: SelectorProps) {
-  const toggleTeam = (team: TeamSelection) => () => {
+type SelectorState = {
+  isOpen: boolean;
+};
+
+class TeamKeyTransactionSelector extends Component<SelectorProps, SelectorState> {
+  constructor(props: SelectorProps) {
+    super(props);
+
+    let portal = document.getElementById('team-key-transaction-portal');
```

Comment:
You can also just have portal be a memoized prop (like how tooltip does it), or do `useMemo` or `useEffect` if you want to keep this a functional component. Up to you though.

Reply:
Not certain how I would use hooks here. To my understanding, `useMemo` should not perform side effects, so would it be something like

1. use `useEffect` to create the element
2. use `useMemo` to get the element (would need some kind of dependency here to get the element on create?)

Probably going to leave it as a class component since the intent seems clearer than with hooks.

Reply:
Yeah no worries, was just saying it's possible. Off the cuff, a single dependency-less use effect could create the portal el  and listeners and remove them on unmount. But totally agree, no reason to use hooks everywhere, was just mentioning it.